### PR TITLE
[networks] Migrate NPM to `ebpf-manager` library

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Updates the network tracer to use the https://github.com/DataDog/ebpf-manager library, which utilizes the upstream `cilium/ebpf` repo rather than a fork of it.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/ebpf/probe/tcp_queue_length.go
Comment (derekwbrown) [inline pkg/collector/corechecks/ebpf/probe/tcp_queue_length.go:23]: Take this fwiw..
I would suggest using something like `ebpfmgr` or something for the library alias (and maybe `ciliumebpf` for the other).
We have so many `manager` types/libs/things in the code, that `manager` is quite ambiguous.  If you call it `ebpfmgr` or something, then the `manager.foo` reference becomes much more obvious when it's `ebpfmgr.foo` in the code.

**Review by derekwbrown — APPROVED** [review]
Approved,  I'll leave it to you the naming convention.

**Comment by Gui774ume** [comment]
Hey @p-lambert 👋🏻 
Could we wait until we merge https://github.com/DataDog/datadog-agent/pull/10094 to merge your PR ? (we're waiting on the CI, so it's only a matter of minutes 😉  ). Otherwise it will break our builds ... that said I don't understand how the CI passed on your PR since the `ExcludedSections` / `ExcludedFunctions` is a breaking change of the lib 🤔

**Comment by brycekahle** [comment]
> that said I don't understand how the CI passed on your PR

@Gui774ume It works because this PR has the same changes as #10094 😁

**Comment by Gui774ume** [comment]
>  @Gui774ume It works because this PR has the same changes as #10094 😁

That makes sense 😆  Haha sorry I didn't see it, and now that we've merged it, there is small conflict on your PR 😅
Anyway, all good from CWS 👍🏻

**Review by Gui774ume — APPROVED** [review]

File: pkg/collector/corechecks/ebpf/probe/tcp_queue_length.go
Comment (L3n41c) [reply pkg/collector/corechecks/ebpf/probe/tcp_queue_length.go:23]: FWIW, I tend to agree with this remark.
But I understand that `DataDog/ebpf-manager` has been aliased with `manager` because it’s a drop-in replacement of the previously used `DataDog/ebpf/manager` which wasn’t aliased.

**Review by L3n41c — APPROVED** [review]
LGTM for the files owned by @DataDog/container-integrations.

File: pkg/collector/corechecks/ebpf/probe/tcp_queue_length.go
Comment (derekwbrown) [reply pkg/collector/corechecks/ebpf/probe/tcp_queue_length.go:23]: true; however, it was ambiguous before, too.  And since we're changing it, we may as well improve that facet as well.

**Pull request merged**
